Utilization on Windows 10

Hi,

it is a great idea for the wrapper. I'm already so used to gradlew - so naturally, I found your repository for my Golang learning journey.

I'm testing your solution on Windows 10. It works but in a strange way.

My example (terminal - Git bash from inside of Visual Studio Code):
MINGW64 /c/prj/go-tutorial-1/create-go-module/hello
$ ../gow.cmd run hello.go
find: ‘/i’: No such file or directory
find: ‘x86’: No such file or directory
go: cannot find GOROOT directory: "C:\prj\go-tutorial-1\create-go-module\.go\wrapper\go"

but Go sources are actually downloaded and unzipped fine.

Then for the run command I have to use 'gow' (bash version) because on 'gow.cmd' I got:
MINGW64 /c/prj/go-tutorial-1/create-go-module/hello
$ ../gow.cmd run hello.go
go: cannot find GOROOT directory: "C:\prj\go-tutorial-1\create-go-module\.go\wrapper\go"

So the result is:
MINGW64 /c/prj/go-tutorial-1/create-go-module/hello
$ ../gow run hello.go
greetings: empty name
exit status 1

And another thing - I insert the line:
distributionUrl=https://dl.google.com/go/go1.13.10.linux-amd64.tar.gz
into the .go/wrapper/go-wrapper.properties file but the downloaded sources are always in the latest version.

Seems like the project is not much active but maybe it is not that much effort to fix those? If you'd like, I'm ready to help to test it.

Hi @furkilic
tl;dr - great job - it worked flawlessly - first try!:)

My quick tests
```
PBANACH@LCE43761 MINGW64 /c/prj/go-tutorial-1/create-go-module/hello
$ ../gow version
go version go1.13.10 windows/amd64

PBANACH@LCE43761 MINGW64 /c/prj/go-tutorial-1/create-go-module/hello
$ ../gow run hello.go 
map[Darrin:Great to see you, Darrin! Gladys:Great to see you, Gladys! Samantha:Hi, Samantha. Welcome!]
```
or cmd/powershell
```
C:\prj\go-tutorial-1\create-go-module\hello>..\gow.cmd version
go version go1.13.10 windows/amd64

C:\prj\go-tutorial-1\create-go-module\hello>..\gow.cmd run hello.go
map[Darrin:Hail, Darrin! Well met! Gladys:Hail, Gladys! Well met! Samantha:Hi, Samantha. Welcome!]
```

Go is a new language to me. I need to get used to the usage of mods and imports. As in my tutorial example, I have a go.mod file in the hello module depending on greeting module like that
```
module example.com/hello

go 1.16

replace example.com/greetings => ../greetings

require example.com/greetings v0.0.0-00010101000000-000000000000
```
this requires run of the gow/gow.cmd from inside of hello module directory - I mean

```
PBANACH@LCE43761 MINGW64 /c/prj/go-tutorial-1/create-go-module/hello
$ ../gow run hello.go 
```

I understand that it can't be run from the root directory
```
PBANACH@LCE43761 MINGW64 /c/prj/go-tutorial-1/create-go-module
$ ./gow run hello/hello.go
hello\hello.go:7:2: cannot find package "example.com/greetings" in any of:
        C:\prj\go-tutorial-1\create-go-module\.go\wrapper\go\src\example.com\greetings (from $GOROOT)        
        C:\Users\pbanach\go\src\example.com\greetings (from $GOPATH)
```
Same results with standard go approach - run by different go version as an alias
```
PBANACH@LCE43761 MINGW64 /c/prj/go-tutorial-1/create-go-module
$ go1.10.7.exe run hello/hello.go
hello\hello.go:7:2: cannot find package "example.com/greetings" in any of:
        C:\Users\pbanach\sdk\go1.10.7\src\example.com\greetings (from $GOROOT)
        C:\Users\pbanach\go\src\example.com\greetings (from $GOPATH)
```
Am I right assuming that this is normal Go flow, or could you explain this behaviour? Maybe _replace example.com/greetings => ../greetings_ can be defined differently?

And I have a question regarding the GOROOT env variable. Should it be set (as a variable in go-wrapper.properties) to existing go installation directory or just left unset? 
I mean it shouldn't be available in the terminal by _echo $GOROOT_ or _echo %GOROOT%_ Do I understand it right? 

Regards,
Paweł

--- a/.github/workflows/main.yml
+++ b/.github/workflows/main.yml
@@ -19,11 +19,27 @@ jobs:
     - name: Install through go-wrapper
       env:
         GOROOT: ""
-      run: ./gow help
+      run: ./gow version
     - name: No need to install go
       env:
         GOROOT: ""
       run: ./gow help
+  install-old-linux:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: Add old version in go-wrapper.properties
+        env:
+          GOROOT: ""
+        run: echo distributionUrl=https://dl.google.com/go/go1.13.10.linux-amd64.tar.gz > .go/wrapper/go-wrapper.properties
+      - name: Install through go-wrapper
+        env:
+          GOROOT: ""
+        run: ./gow version
+      - name: No need to install go
+        env:
+          GOROOT: ""
+        run: ./gow help
   no-install-linux:
     runs-on: ubuntu-latest
     steps:
@@ -39,13 +55,31 @@ jobs:
     - name: Install through go-wrapper
       env:
         GOROOT: ""
-      run: "gow.cmd help"
+      run: "gow.cmd version"
       shell: cmd
     - name: No need to install go
       env:
         GOROOT: ""
       run: "gow.cmd help"
       shell: cmd
+  install-old-windows:
+    runs-on: windows-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: Add old version in go-wrapper.properties
+        env:
+          GOROOT: ""
+        run: echo distributionUrl=https://dl.google.com/go/go1.13.10.windows-amd64.zip > .go/wrapper/go-wrapper.properties
+      - name: Install through go-wrapper
+        env:
+          GOROOT: ""
+        run: "gow.cmd version"
+        shell: cmd
+      - name: No need to install go
+        env:
+          GOROOT: ""
+        run: "gow.cmd help"
+        shell: cmd
   no-install-windows:
     runs-on: windows-latest
     steps:

--- a/README.md
+++ b/README.md
@@ -81,6 +81,10 @@ You can change its version by setting the `distributionUrl` in
 ```bash
 distributionUrl=https://dl.google.com/go/go1.13.10.linux-amd64.tar.gz
 ```
+```bash
+distributionUrl=https://dl.google.com/go/go1.13.10.windows-amd64.zip
+```
+> :warning: **Be very careful here to download the correct url for windows and linux** 
 
 ## Using Basic Authentication for Distribution Download
 
@@ -99,3 +103,19 @@ options:
 
 If you already have a local installation and your `GOROOT` environment variable points 
 to it, Go Wrapper will use it.
+
+## FAQ
+
+#### Should i use `gow` or `gow.cmd`
+
+Only on Windows and the `Command Prompt` or `PowerShell` use `./gow.cmd`
+
+For all other options (_Windows with `GitBash` or `Cygwin`, Linux, MacOs.._) use `./gow`
+
+#### Nothing is happening when I put an older `distributionUrl` in `.go/wrapper/go-wrapper.properties`
+
+First of all ensure that  the `distributionUrl` matches the OS your are working on.
+
+Ensure that there is not an already avalaible `GOROOT` in your Environment Variables
+
+In case there is a previous Installation with `gow`, manually delete `.go/wrapper/tmp` and `.go/wrapper/go`

--- a/gow
+++ b/gow
@@ -131,7 +131,9 @@ unpack_archive(){
   else
     tar -C "$goWrapperPath" -zxvf "$goZipPath"
   fi
-  rm "$goVersionPath"
+  if [ -f "$goVersionPath" ]; then
+    rm "$goVersionPath"
+  fi
   date > "$goWrapperDate"
 }
 
@@ -151,8 +153,8 @@ get_local_go() {
         fi
         mkdir -p "$goTmpPath"
         if [ -f "$goWrapperProperties" ]; then
-            while IFS="=" read key value; do
-              case "$key" in (distributionUrl) goUrl="$value"; break ;;
+            while IFS="=" read key value  || [ "$key" ]; do
+              case "$key" in (distributionUrl) goUrl=`echo "$value" | sed 's/ *$//g'`; break ;;
               esac
             done < "$goWrapperProperties"
         fi
@@ -188,9 +190,9 @@ get_local_go() {
             if [ -z "$goUrl" ]; then
                 if [ ! -f "$goVersionPath" ]; then
                     if [ -z "$GOW_USERNAME" ] || [ -z "$GOW_PASSWORD" ]; then
-                        curl -o "$goVersionPath" "$goVersionUrl" -f
+                        curl -o "$goVersionPath" "$goVersionUrl" -f -s
                     else
-                        curl --user $GOW_USERNAME:$GOW_PASSWORD -o "$goVersionPath" "$goVersionUrl" -f
+                        curl --user $GOW_USERNAME:$GOW_PASSWORD -o "$goVersionPath" "$goVersionUrl" -f -s
                     fi
                 fi
                 go_version=`retrieve_version "$goVersionPath"`
@@ -219,7 +221,7 @@ if [ -z "$GOROOT" ]; then
     GOROOT="$goInstallPath"
 fi
 
-# For Cygwin, switch paths to Windows format before running java
+# For Cygwin, switch paths to Windows format before running go
 if $cygwin; then
   [ -n "$GOROOT" ] &&
     GOROOT=`cygpath --path --windows "$GOROOT"`

--- a/gow.cmd
+++ b/gow.cmd
@@ -77,8 +77,8 @@ cd "%EXEC_DIR%"
 :endDetectBaseDir
 
 set GO_WRAPPER_PATH="%GO_PROJECTBASEDIR%\.go\wrapper"
-set GO_WRAPPER_PROPERTIES="%GO_PROJECTBASEDIR%\.go\wrapper\go-wrapper.properties"
-set GO_INSTALL_PATH="%GO_PROJECTBASEDIR%\.go\wrapper\go"
+set GO_WRAPPER_PROPERTIES=%GO_PROJECTBASEDIR%\.go\wrapper\go-wrapper.properties
+set GO_INSTALL_PATH=%GO_PROJECTBASEDIR%\.go\wrapper\go
 set GO_TMP_PATH="%GO_PROJECTBASEDIR%\.go\wrapper\tmp"
 set GO_WRAPPER_DATE="%GO_PROJECTBASEDIR%\.go\wrapper\go\go.date"
 set GO_VERSION_URL="https://golang.org/dl/"

